packages/did/src/DidDetails/LightDidDetails.utils.ts

```diff
@@ -7,55 +7,111 @@
 
 import { encode as cborEncode, decode as cborDecode } from 'cbor'
 
-import { SDKErrors } from '@kiltprotocol/utils'
 import { base58Decode, base58Encode } from '@polkadot/util-crypto'
 
-import type { LightDidCreationDetails } from '../types.js'
+import type {
+  DidServiceEndpoint,
+  NewDidEncryptionKey,
+  NewDidVerificationKey,
+  SubmittableExtrinsic,
+} from '@kiltprotocol/types'
+import { EncryptionKeyType, VerificationKeyType } from '@kiltprotocol/types'
+
+import { SDKErrors } from '@kiltprotocol/utils'
+
 import { parseDidUri } from '../Did.utils.js'
 
 const ENCRYPTION_KEY_MAP_KEY = 'e'
 const SERVICES_KEY_MAP_KEY = 's'
 
-export enum LightDidSupportedSigningKeyTypes {
-  ed25519 = 'ed25519',
-  sr25519 = 'sr25519',
-}
+// Ecdsa not supported.
+export type LightDidSupportedVerificationKeyType =
+  | VerificationKeyType.Ed25519
+  | VerificationKeyType.Sr25519
 
-export enum LightDidSupportedEncryptionKeyTypes {
-  x25519 = 'x25519',
+// Ecdsa not supported.
+export function getEncodingForVerificationKeyType(
+  type: VerificationKeyType
+): string | undefined {
+  switch (type) {
+    case VerificationKeyType.Sr25519:
+      return '00'
+    case VerificationKeyType.Ed25519:
+      return '01'
+    default:
+      return undefined
+  }
 }
-
-const supportedEncryptionKeyTypes = new Set<string>([
-  ...Object.keys(LightDidSupportedEncryptionKeyTypes),
-])
-
-const EncodingForSigningKeyType = {
-  [LightDidSupportedSigningKeyTypes.sr25519]: '00',
-  [LightDidSupportedSigningKeyTypes.ed25519]: '01',
+export function getVerificationKeyTypeForEncoding(
+  encoding: string
+): LightDidSupportedVerificationKeyType | undefined {
+  switch (encoding) {
+    case '00':
+      return VerificationKeyType.Sr25519
+    case '01':
+      return VerificationKeyType.Ed25519
+    default:
+      return undefined
+  }
 }
 
-export function getEncodingForSigningKeyType(
-  keyType: string
-): string | undefined {
-  return EncodingForSigningKeyType[keyType]
-}
+const supportedEncryptionKeyTypes = new Set(Object.values(EncryptionKeyType))
 
-const SigningKeyTypeFromEncoding = {
-  '00': LightDidSupportedSigningKeyTypes.sr25519,
-  '01': LightDidSupportedSigningKeyTypes.ed25519,
+/**
+ * A new public key specified when creating a new light DID.
+ *
+ * Currently, a light DID does not support the use of an ECDSA key as its authentication key.
+ */
+export type NewLightDidAuthenticationKey = Omit<
```

Comment:
C: Feels like this should live in some kind of `type` file

Reply:
https://github.com/KILTprotocol/sdk-js/pull/473/commits/34556a067e8965b6fec24914c6d216b01e971911